packages/dataviews/src/dataviews.tsx

```diff
@@ -38,10 +38,12 @@ interface DataViewsProps< Item extends AnyItem > {
 		totalPages: number;
 	};
 	supportedLayouts: string[];
+	selection?: string[];
+	setSelection?: ( selection: string[] ) => void;
 	onSelectionChange?: ( items: Item[] ) => void;
 }
 
-const defaultGetItemId = ( item: AnyItem ) => item.id;
+const defaultGetItemId = ( item: AnyItem ) => item.id + '';
```

Comment:
Is this needed?

Reply:
Yes it is needed, it is making sure the ID returned is a string, in some cases, it was an integer that made the comparisons fail. The selections have a type of string[] and previously many times it was an integer array.

Reply:
Hi @youknowriad I applied your suggestion.